src/main/crc.ts

```diff
@@ -0,0 +1,54 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { crc32 } from "crc";
+import * as yauzl from "yauzl";
+
+type TCrcFile = [string, string];
+
+export async function extractCrc32OnZip(filePath: string) {
+    const fileArray: TCrcFile[] = [];
+    return new Promise<string>((resolve, reject) => {
+
+        yauzl.open(filePath, { lazyEntries: true }, (err, data) => {
+            if (err) {
+                reject(err);
+            }
+            data.on("error", (e) => reject(e));
+            data.readEntry();
+            data.on("entry", (entry: yauzl.Entry) => {
+                fileArray.push([entry.fileName, (entry.crc32 || 0).toString(16)]);
+                data.readEntry();
+            });
+            data.on("end", () => {
+                const sortArray = fileArray.sort((a, b) => a[0] === b[0] ? 0 : a[0] < b[0] ? -1 : 1);
```

Comment:
Based on my tests with a large EPUB set, sorting is not necessary.
See https://github.com/readium/readium-desktop/pull/793#issuecomment-542363462

Reply:
Based on my tests on Thorium, I think is necessary to sorting. We cannot be sure that the order received on entry event is correct

Reply:
I have extensively batch-tested (and stress-tested) this assumption in the `r2-utils-js` using an automated benchmarking script, so I beg to differ :)
Do you have data to backup this claim, or just anecdotal evidence?

Reply:
Sorting is guaranteed with `yauzl` despite the asynchronous `entry` event, thanks to the `readEntry()` function call. So we can eliminate the string sorting :)